experimental/dds/tree2/src/shared-tree/sharedTree.ts

```diff
@@ -202,6 +216,41 @@ export class SharedTree
 			events: this._events,
 			removedTrees,
 		});
+		this.viewForSummarization = this.view;
+
+		// If the attach summary is requested when there's a pending transaction, we need to load from the state before
+		// that transaction was open (else other clients will load from the intermediate state of the transaction).
+		// We guarantee this by forking the view we should use to summarize if a transaction is opened while detached.
+		let activeTransactionCount = 0;
+		const beginTransaction = () => {
+			if (activeTransactionCount === 0) {
+				this.viewForSummarization = this.view.fork();
```

Comment:
Even with ChunkedForest, this will have add decent cost to attach summaries considering AnchorSet and it's dependents are not forkable and will be recreated for the summary (I'm admittedly not sure how much we care about large attaches).

Reply:
Large attaches are definitely something we should care about: copy-paste of a large tree will trigger that. The thing is, that still doesn't hit the potentially problematic perf characteristics here: this only adds sizeable cost when someone creates a reasonably significant sized tree *before* then starting a transaction all while detached. It's a pretty niche case, the only real-life usage I can think of is if someone is pasting in some new content which partially requires async-fetched data, but for some reason they want to allow collaborative editing before fully loading that data. Everywhere else, you just shouldn't use transactions while detached.

That said, I added a test to makes sure we deregister this fork upon attaching: I agree the cost is unpalatable if paid more frequently.